aten/src/ATen/native/cuda/Shape.cu

```diff
@@ -240,6 +270,27 @@ Tensor cat_cuda(TensorList inputs, int64_t dimension) {
   return out;
 }
 
+inline c10::MemoryFormat compute_output_memory_format(const TensorList &inputs) {
+  c10::optional<c10::MemoryFormat> format = c10::nullopt;
+  for (auto &t : inputs) {
+    auto f = t.suggest_memory_format();
```

Comment:
what happens if one of the tensors is undefined?

Reply:
Do we need to check if tensors are defined? There is no such check in shape computation.

Reply:
I don't think we should do defined check in this function, but we might need to do it in `cat_out`.

Reply:
I don't really know if it's needed here, I've seen it below in `allContiguous`. Would `suggest_memory_format` even work on undefined tensor?

Reply:
No, it won't. It calls `sizes()` which is checked for undefined tensor.